### PR TITLE
ChibiOS: use more consistent naming for power VALID pins

### PR DESCRIPTION
**Pull request by tridge**
this uses _VALID for an active high pin and _nVALID for an active low pin. It uses _OC for active high overcurrent and _nOC for active low overcurrent

ping @pkocmoud @bugobliterator

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/mRoX21-777/hwdef.dat [hunk 1]
```diff
@@ -238,7 +238,7 @@ PB14 SPI2_MISO SPI2
 PB15 SPI2_MOSI SPI2
 
 # this input pin is used to detect that power is valid on USB
-PC0 VBUS_VALID INPUT
+PC0 VBUS_VALID INPUT PULLDOWN
```
Comment (pkocmoud) [inline libraries/AP_HAL_ChibiOS/hwdef/mRoX21-777/hwdef.dat]:
```suggestion
PC0 VBUS_VALID INPUT PULLDOWN -> PC0 VBUS_nVALID INPUT PULLUP
```

**Comment by pkocmoud** [comment]
Pixracer tested
x2.1-777 corrected and tested
x2.1 tested and needs to be corrected

**Comment by tridge** [comment]
>  x2.1 tested and needs to be corrected

X2.1 is just copied from fmuv3 at the moment. Does it differ from fmuv3 power or is fmuv3 wrong?

**Comment by tridge** [comment]
>  x2.1 tested and needs to be corrected

@pkocmoud can you tell me what needs correcting?

File: libraries/AP_HAL_ChibiOS/AnalogIn.cpp [hunk 2]
```diff
@@ -366,44 +366,113 @@ void AnalogIn::update_power_flags(void)
...
-    if (!palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID)) {
+    /*
+      primary "brick" power supply valid pin. Some boards have this
+      active high, some active low. Use nVALID for active low, VALID
+      for active high
+    */
+#if defined(HAL_GPIO_PIN_VDD_BRICK_VALID)
+    if (palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID) == 1) {
+        flags |= MAV_POWER_STATUS_BRICK_VALID;
+    }
```
Comment (bugobliterator) [inline libraries/AP_HAL_ChibiOS/AnalogIn.cpp:377]: This looks like something for which we can generate macro for, which converts HAL_GPIO_PIN_VDD_BRICK_VALID into (palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID) == 1) or (palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID) == 0) making it even more concise and readable.

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/AnalogIn.cpp:377]: we could, but for now I just want to know if I have broken anything. Can you check it is correct for the Hex boards?

**Comment by pkocmoud** [comment]
@tridge Please disregard my comment, this tests fine on the x2.1.

**Comment by tridge** [comment]
> Tested all Hex/Proficnc boards, looks good.

thanks! I'll merge

**Pull request merged**